src/lib.rs

```diff
@@ -342,22 +759,83 @@ mod tests {
             "oauth_timestamp=1471445561&",
             "oauth_version=1.0",
         ]
-        .iter()
-        .cloned()
-        .collect::<String>();
+            .iter()
+            .cloned()
+            .collect::<String>();
         let encoded_query = [
             "oauth_consumer_key%3Dkey%26",
             "oauth_nonce%3Ds6HGl3GhmsDsmpgeLo6lGtKs7rQEzzsA%26",
             "oauth_signature_method%3DHMAC-SHA1%26",
             "oauth_timestamp%3D1471445561%26",
             "oauth_version%3D1.0",
         ]
-        .iter()
-        .cloned()
-        .collect::<String>();
+            .iter()
+            .cloned()
+            .collect::<String>();
 
         assert_eq!(encode(method), "GET");
         assert_eq!(encode(uri), encoded_uri);
         assert_eq!(encode(&query), encoded_query);
     }
 }
+
+use log::warn;
+
+/// Assumptions:
+/// 1. Keys are distinct
+/// 2. `warn!` is in scope, either from `log` or `tracing` etc.
+///
+/// Arguments:
+/// 1. &str Key to search
+/// 2. bool Whether to (unstable-y) sort the return value (for reproducibility)
+/// 3+. Variadic! `&str` The names of the keys. (If put more than existing, or invalid then error might happen
+///    because we are looking for all provided keys.)
+#[macro_export]
+macro_rules! parse_query_string {
```

Comment:
I tried it but looks like the transmute is blocked by https://github.com/rust-lang/rust/issues/61956 (with impl in https://github.com/rust-lang/rust/pull/86281) as I get:
```
cannot transmute between types of different sizes, or dependently-sized types E0512 Note: source type: `[Option<(&str, &str)>; N]` (this type does not have a fixed size) Note: target type: `[(&str, &str); N]` (this type does not have a fixed size)
```

I'd be fine with keeping it as is until the issue gets resolved.

Reply:
https://gist.github.com/JakubKoralewski/381f5e858e3270cdd14e0cf871121c74

Reply:
`transmute_copy` can be used.
Moreover, in this case it is better to use `MaybeUninit` instead of` Option` as in [this commit].

[this commit]: https://github.com/gifnksm/oauth-client-rs/commit/69ba568ec4dfc6d4aaa976d0dd6fbd05506f3331